### ISSUE TITLE
Support MediaWiki Wikitext link format.

### ISSUE THREAD

**Issue by fidergo-stephane-gourichon** [asamuzaK/url2clipboard#34]
Basically, it looks like:

`[https://mediawiki.org MediaWiki]`

User introduction [Help:Links - MediaWiki, section "External links to pages at other websites"](https://www.mediawiki.org/wiki/Help:Links#External_links "Help:Links - MediaWiki")

Links to somehow formal spec on https://www.mediawiki.org/wiki/Specs/wikitext/1.0.0 including automated big test spec [parserTests.txt](https://github.com/wikimedia/mediawiki/blob/master/tests/parser/parserTests.txt "parserTests.txt").

**Comment by asamuzaK** [comment]
Will this be the right way to escape brackets?
```
[http://example.com ![content with brackets!]]
```

**Comment by asamuzaK** [comment]
ref [Help:Wikitext examples - Meta](https://meta.wikimedia.org/wiki/Help:Wikitext_examples "Help:Wikitext examples - Meta")

**Comment by asamuzaK** [comment]
[Help:Wikitext examples - Meta](https://meta.wikimedia.org/wiki/Help:Wikitext_examples "Help:Wikitext examples - Meta")

> You can put formatting around a link.
Example: ''[[Wikipedia]]''.

But formatting not allowed inside?

**Comment by asamuzaK** [comment]
And another
How will IPv6 URL be handled?
```
[http://[::1]/ localhost]
```

**Comment by fidergo-stephane-gourichon** [comment]
> Will this be the right way to escape brackets?

I don't know all the answers. You can try on the mediawiki (or wikipedia) website.

Regarding formatting inside, my guess is it's not important in the dedicated use case of an encyclopedia. 
If you have more doubt about any char (e.g. brackets) I guess you can just strip them and be okay.
Also, you can leave starts, underscore etc if they don't have their formatting effects. They'll just be rendered as themselves and be okay, since it's one of the goals of wiki markup.

> How will IPv6 URL be handled?
> `[http://[::1]/ localhost]`

Tried on mediawiki.org preview. Seems to work very well:

`<a rel="nofollow" class="external text" href="http://[::1]/">localhost</a>`

**Comment by asamuzaK** [comment]
Tested on [Wikipedia:Sandbox - Wikipedia](https://en.wikipedia.org/wiki/Wikipedia:Sandbox "Wikipedia:Sandbox - Wikipedia")

```
[http://example.com [content with brackets]]
-> <a rel="nofollow" class="external text" href="http://example.com">[content with brackets</a>]

[http://example.com &#91;content with brackets&#93;]
-> <a rel="nofollow" class="external text" href="http://example.com">[content with brackets]</a>

[http://example.com content with ''italic'' markup]
-> <a rel="nofollow" class="external text" href="http://example.com">content with <i>italic</i> markup</a>
```

**Comment by asamuzaK** [comment]
So, it seems that brackets and characters used for markups need to be converted...
Is there a quick reference or a list which character is used for markup?
I don't want to look into all MediaWiki Help:foo docs...

**Comment by asamuzaK** [comment]
Picked up from [Help:Wikitext examples - Meta](https://meta.wikimedia.org/wiki/Help:Wikitext_examples "Help:Wikitext examples - Meta")
```
[]'~<>{}=*#;:\-|
```

**Comment by asamuzaK** [comment]
Try build [Version 0.19.0a1](https://addons.mozilla.org/en-US/firefox/addon/url2clipboard/versions/beta "Copy URL To Clipboard :: Versions :: Add-ons for Firefox")

**Comment by asamuzaK** [comment]
Released v0.19.0 on AMO.
You need to update yourself from beta to release channel.
Thanks.
